build/compile.sh

```diff
@@ -171,22 +171,33 @@ build() {
 
   pushd $PY27
 
-  local compile_readline
-  local link_readline
+  if is_macos; then
+    cp pyconfig.macos.h pyconfig.h
+  else
+    cp pyconfig.linux.h pyconfig.h
```

Comment:
I wouldn't call this pyconfig.linux.h, because in theory it should work for all platforms.  I am accepting pyconfig.macos.h for expedience, as a hack, but ideally these things would converge and it would all be detected at `configure` time.

So maybe just call it pyconfig.default.h and pyconfig.darwin.h.  I think darwin is better because that's what `uname` says.

Or it would be even better if the headers can be merged, but I don't know how much work that is.

Reply:
I decided to take a stab at unifying the headers and adding proper feature detection where needed 😁